Bump dlib to 1.2.4.

josecv opened the pull request:
This'll include some perf improvements for log levels > debug

Signed-off-by: Jose Cortes <josecortes@datawire.io>

## Description

A few sentences describing the overall goals of the pull request's commits.

## Checklist

<!--
  Please review the requirements for each checkbox, and check them
  off (change "[ ]" to "[x]") as you verify that they are complete.
-->

 - [ ] I made sure to update `./CHANGELOG.md`.
 - [ ] I made sure to either submit a docs PR, or tell Matt about the necessary documentation changes.
 - [ ] My change is adequately tested.
 - [ ] I updated `DEVELOPING.md` with any any special dev tricks I had to use to work on this code efficiently.

josecv committed:
Bump dlib to 1.2.4.

This'll include some perf improvements for log levels > debug

Signed-off-by: Jose Cortes <josecortes@datawire.io>

thallgren reviewed (approved):
Looks good to me. The `pkg/client/logging/discard.go` has moved to `pkg/log` though, so you'll need to resolve that.

josecv committed:
Run windows commands in their own process group.

This is to prevent failures from dexec around windows not being able to
hard-cancel commands not in their own process group

Signed-off-by: Jose Cortes <josecortes@datawire.io>

josecv committed:
Update OPENSOURCE.md

Signed-off-by: Jose Cortes <josecortes@datawire.io>

josecv committed:
Set debug logging for the connector in tests

Signed-off-by: Jose Cortes <josecortes@datawire.io>

thallgren commented on pkg/dpipe/dpipe_windows.go:
```diff
@@ -23,6 +23,12 @@ type processInfo struct {
 	exe  string
 }
 
+func prepareCmd(cmd *dexec.Cmd) {
```
Given the somewhat thorough peruse of child processes and how they are killed in the right order on Windows, I'm surprised this would be needed.

thallgren commented on pkg/tun/device_windows.go:
```diff
@@ -151,7 +154,12 @@ $job | Receive-Job
 	cmd = dexec.CommandContext(ctx, "ipconfig", "/flushdns")
 	cmd.DisableLogging = true
-	_ = cmd.Run()
+	cmd.SysProcAttr = &windows.SysProcAttr{
```
does `ipconfig` really start subprocesses? If not, why is this needed?

thallgren commented on pkg/tun/device_windows.go:
```diff
@@ -142,6 +142,9 @@ $job | Receive-Job
 `, t.interfaceIndex, domain)
 	cmd := dexec.CommandContext(ctx, "powershell.exe", "-NoProfile", "-NonInteractive", pshScript)
 	cmd.DisableLogging = true // disable chatty logging
+	cmd.SysProcAttr = &windows.SysProcAttr{
```
Does `powershell.exe` start subprocesses? If not, then why is a process group significant?

thallgren commented on pkg/client/scout/os_metadata_windows.go:
```diff
@@ -6,13 +6,18 @@ import (
 func getOsMetadata(ctx context.Context) map[string]interface{} {
 	cmd := dexec.CommandContext(ctx, "systeminfo")
 	cmd.DisableLogging = true
+	cmd.SysProcAttr = &windows.SysProcAttr{
```
Same here. Don't think the `systeminfo` starts additional processes, so I don't understand the need for a process group. On the contrary actually. Don't we want this to die if the parent dies?

thallgren reviewed (changes requested):
It's very possible that I've misunderstood something regarding Windows and how it handles process-groups, but it seems to me that adding specific behaviour to all (or most) invocations of a command to rectify a windows problem (or regression in dlib?) should belong in `dlib` rather than being repeated over and over again in our code.

josecv committed:
Address PR feedback

Signed-off-by: Jose Cortes <josecortes@datawire.io>

thallgren commented:
Closing this since the upgrade of dlib was made some time ago.
